src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -1,19 +1,23 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Concurrent;
 using System.Security.Cryptography.X509Certificates;
 
 namespace System.Net.Security
 {
     public partial class SslStreamCertificateContext
     {
         private const bool TrimRootCertificate = true;
+        internal readonly ConcurrentDictionary<int, SafeSslContextHandle> contexts;
 
         private SslStreamCertificateContext(X509Certificate2 target, X509Certificate2[] intermediates, SslCertificateTrust? trust)
         {
             Certificate = target;
             IntermediateCertificates = intermediates;
             Trust = trust;
+            contexts = new ConcurrentDictionary<int, SafeSslContextHandle>();
```

Comment:
Is there a risk this grows unbounded?  What clears out state from it?

Reply:
At the moment we never clear. SslStreamCertificateContext is _not_ IDisposable so this will go away when there is no reference to it. 
In theory, it can get all combinations of [SslProtocolos](https://docs.microsoft.com/en-us/dotnet/api/system.security.authentication.sslprotocols?view=net-5.0).  In practice, I expect this to be one entry e.g. the SslStreamCertificateContext would be one server with given enabled protocols. (and some protocols like Ssl2 & Ssl3 most likely throwing so they would not get ever added) 
I can also possibly make it `ConcurrentDictionary<SslProtocolos, SafeSslContextHandle>` to make it more clear but `SslProtocolos` is used as bitmaps/Flags so I was not sure if that is appropriate.

Reply:
I made update to use SslProtocols instead of int. It should not change the size of behavior but hopefully it will make it more clear.